target/linux/generic/files/drivers/mtd/mtdsplit/mtdsplit_tplink.c

```diff
@@ -105,25 +105,24 @@ static int mtdsplit_parse_tplink(struct mtd_info *master,
 
 	switch (le32_to_cpu(hdr.version)) {
 	case 1:
-		if (be32_to_cpu(hdr.v1.kernel_ofs) != sizeof(hdr))
-			return -EINVAL;
-
-		kernel_size = sizeof(hdr) + be32_to_cpu(hdr.v1.kernel_len);
+		kernel_offset = be32_to_cpu(hdr.v1.kernel_ofs);
+		kernel_size = kernel_offset + be32_to_cpu(hdr.v1.kernel_len);
 		rootfs_offset = be32_to_cpu(hdr.v1.rootfs_ofs);
+		rootfs_size = be32_to_cpu(hdr.v1.rootfs_len);
 		break;
 	case 2:
 	case 3:
-		if (be32_to_cpu(hdr.v2.kernel_ofs) != sizeof(hdr))
-			return -EINVAL;
-
-		kernel_size = sizeof(hdr) + be32_to_cpu(hdr.v2.kernel_len);
+		kernel_offset = be32_to_cpu(hdr.v2.kernel_ofs);
+		kernel_size = kernel_offset + be32_to_cpu(hdr.v2.kernel_len);
```

Comment:
Shouldn't the kernel_size == kernel_len _if_ **kernel_offset** == kernel_ofs?

Reply:
hmm no. That's a left-over bit of an overly aggressive sed. Will fix, thanks